docs/queryworkers/attribute-based-access.md

```diff
@@ -0,0 +1,260 @@
+}
+```
+
+In this example, customer 1 is partner 1 and customer 2 is partner 2.
+
+## Example 1
+
+In this example, partners can see their own items but cannot update them.
+
+The query worker `PartnerItems` displays data from a collection to which the active user can access:
+
+```sql
+FOR doc IN items
+    FILTER doc.partner == CURRENT_APIKEY_ATTRIBUTE("partner")
+    RETURN doc
+```
+
+When Partner 1 wants to view their data, they can run a command with their API key similar to the following:
+
+```sql
+curl -X 'POST' \\
+  '<https://fulfillment.eng.macrometa.io/_fabric/_system/_api/restql/execute/PartnerItems>' \\
+  -H 'accept: application/json' \\
+  -H 'Content-Type: application/json' \\
+  -H 'Authorization: apikey <PARTNER 1 API KEY>' \\
+  -d '{ "bindVars": {}}'
+```
+
+Note that in the results, only data with the `partner1` attribute displays:
+
+```sql
+{
+   { "item": "hammer", "price": 5.99, "count": 55, "partner": "partner1" },
+   { "item": "screw driver", "price": 3.99, "count": 15, "partner": "partner1" },
+}
+```
+
+## Example 2
+
+In this example, partners can see their orders and update them when they are filled.
+
+The query worker `PartnerOrdersUpdate` updates data in the collection if the active user has access:
+
+```sql
+FOR doc IN orders
+    FILTER doc.partner == CURRENT_APIKEY_ATTRIBUTE("partner") AND
+            doc.customer == @customer
+    UPDATE doc WITH {"ordered": doc.ordered + 1} IN orders
+    RETURN NEW
+```
+
+When Partner 1 wants to update their data, they can run a command with their API key similar to the following:
+
+```
+curl -X 'POST' \\
+    '<https://fulfillment.eng.macrometa.io/_fabric/_system/_api/restql/execute/PartnerOrdersUpdate>' \\
+    -H 'accept: application/json' \\
+    -H 'Content-Type: application/json' \\
+    -H 'Authorization: apikey acme_key.897b6e...' \\
+    -d '{ "bindVars": { "customer": 1}}'
+```
+
+The displayed results show the added data:
+
+```sql
+{
+   { "customer": 1, "item": "hammer", "ordered": 2, "partner": "partner1" },
+}
+```
+
+On the other hand, if Partner 1 tries to run the same command to edit data belonging to Partner 2:
+
+```
+curl -X 'POST' \\
+    '<https://fulfillment.eng.macrometa.io/_fabric/_system/_api/restql/execute/PartnerOrdersUpdate>' \\
+    -H 'accept: application/json' \\
+    -H 'Content-Type: application/json' \\
+    -H 'Authorization: apikey <PARTNER 1 API KEY>' \\
+    -d '{ "bindVars": { "customer": 2}}'
+```
+
+The result is empty because the attempt failed:
+
+```sql
+{
+   {}
+}
+```
+
+## Example 3
+
+In this example, partners attempt to view employee data and are blocked because they do not have proper access.
+
+The name of this query worker is `Employees`:
+
+```sql
+FOR doc IN employees
+    FILTER CURRENT_APIKEY_ATTRIBUTE("partner") == null
+    RETURN doc
+```
+
+The result is empty because API keys with the `partner` attribute do not have access to the `employees` collection:
+
+```sql
+{
+   {}
+}
+```
+
+## Example 4
```

Comment:
Attribute View and Edit Permissions

...okay, that one is not great. You're right, this is confusing. Also, are these query workers each working individually, or are they all building the same API, so all existing together? I thought the latter, but now I am not so sure.

Reply:
I believe they're all being run at the same time.  @joubert-macrometa can you confirm please?